### PR TITLE
Vcd 11 - Provide a Docker Compose file

### PR DESCRIPTION
**Pull request by monda07**
Add a docker-compose.yml file to simply all operations for those that are aware of docker-compose
Fix a minor problem with the editing of the config.dat file that will cause imadb queries to fail
Clean up the vector tarball to save space in the image

### CODE REVIEW

**Comment by clach04** [comment]
👍 Looks good, I'd like to get this in.

As agreed in #11, this PR is on hold until #11 is submitted and then @monda07 will resolve conflicts so this can then be merged. I will be happy to approve and merge unless anyone has objections.

**Comment by clach04** [comment]
#11 now merged, so this now conflicts.

**Comment by monda07** [comment]
Master merged with some additional fixes

**Comment by clach04** [comment]
Making sure I understand this:
* Add Visual Studio Code default registry path
* Disable the management server in the container
* Adds Docker compose file
* Misc doc updates

Do the ports that get exposed include the management server port? Does that need updating? We probably should have a note in the readme about that.

Looking good. Thoughts @fgropengieser?

File: Dockerfile
Comment (fgropengieser) [inline Dockerfile:47]: I do not understand, why this change is required.

File: .vscode/settings.json
Comment (fgropengieser) [inline .vscode/settings.json]: From my point of view, VSCode settings should not go into the repository.

File: docker-compose.yml
Comment (fgropengieser) [inline docker-compose.yml]: Does this imply that we have the image pushed to our private Actian docker registry? If so, it should go into documentation. Another question: I thought, we do not offer community edition anymore. So this should then be removed, right?

File: .vscode/settings.json
Comment (monda07) [reply .vscode/settings.json]: This is where Visual Studio put the file by default.   I don't use VS so I have no idea what the best practices are.  This was just the result of using VS on this project.

File: docker-compose.yml
Comment (monda07) [reply docker-compose.yml]: This file is meant to be an example mostly and no, there was no intention to use hub.docker.com at this time.  Since this project builds the image, it will be local to the users environment.  Personally I think it would be cool to have a Vector image publicly available on Docker Hub if we could deal with the licensing issues.

In any case, :community is a bug because the Dockerfile was modified to remove community references and I neglected to remove it here after the merge.

File: Dockerfile
Comment (monda07) [reply Dockerfile:47]: Which one?  ;-)
-isaved causes the in-place edit by sed to retain a backup file of the config.dat just because that seems like a good idea, especially in diagnostic or debugging.
The management server is a massive component and should not be included in a container.  I was attempting to make the container at least slightly more svelte.....

Comment (fgropengieser) [reply Dockerfile:47]: I am not familiar with all Ingres components, but does the Vector container work without the mgmtsvr?

File: .vscode/settings.json
Comment (fgropengieser) [reply .vscode/settings.json]: Yeah, they are put into the hidden folder in you local working copy. However, you should not add it to the commit list. Rather you should add it to gitignore.

Comment (clach04) [reply .vscode/settings.json]: In that case, I agree remove (and potentially add to `.gitignore`. I was assuming you felt strongly about including it, I wasn't keen but not enough to reject.

File: Dockerfile
Comment (clach04) [reply Dockerfile:47]: Yes, Vector will work without mgmtsvr - its only used by the Director (client).

Should someone want to add/restore this they can do so by tweaking the Dockerfile (i.e. not worth making a parameter).

I'm in favor of this

**Comment by clach04** [comment]
👍 lgtm :)

File: Dockerfile
Comment (monda07) [reply Dockerfile:47]: A better long term solution would be to make the management server a stand alone container but yes, it is easy for someone to modify the Dockerfile if that was an issue. (There is no scenario I can think of where someone would do this ;-)

**Comment by monda07** [comment]
Updates pushed.

**Review by fgropengieser — APPROVED** [review]

**Review by clach04 — APPROVED** [review]

**Comment by clach04** [comment]
squashed, customized commit, merged

**Comment by monda07** [comment]
Thanks everyone for the great feedback!

**Pull request merged**
